glue_jupyter/bqplot/common/tools.py

```diff
@@ -153,7 +153,21 @@ def update_selection(self, *args):
                 x = self.interact.selected_x
                 y = self.interact.selected_y
                 theta = None if self._roi is None else self._roi.theta
-                roi = RectangularROI(xmin=min(x), xmax=max(x), ymin=min(y), ymax=max(y), theta=theta)  # noqa: E501
+                x_min = min(x)
+                x_max = max(x)
+                y_min = min(y)
+                y_max = max(y)
+                # Avoid drawing invisible shape.
+                if isinstance(self._roi, RectangularROI):
+                    dx = abs(x_max - x_min)
+                    dy = abs(y_max - y_min)
+                    if dx < 1:
+                        x_min = self._roi.xmin
+                        x_max = self._roi.xmax
+                    if dy < 1:
+                        y_min = self._roi.ymin
+                        y_max = self._roi.ymax
```

Comment:
```suggestion
                # Avoid drawing invisible shape.
                x_min = min(min(self.interact.selected_x), max(self.interact.selected_x) - 1.0)
                x_max = max(max(self.interact.selected_x), min(self.interact.selected_x) + 1.0)
                y_min = min(min(self.interact.selected_y), max(self.interact.selected_y) - 1.0)
                y_max = max(max(self.interact.selected_y), min(self.interact.selected_y) + 1.0)
```
Simpler to catch too small sizes already here imo; also the later check for `dx < 1` has the disadvantage of abruptly snapping back to the original size, which seems less intuitive to handle.
In principle an offset of 0.5 above should be enough, but I could not reliably keep it from resizing below 1 with that.

Reply:
But I want it to snap back to original size. The actual bug we encounter is explained in the original post above and repeated here:

```
selected_x: [34.543987, 35.289345]
selected_y: [18.55497, 17.809614]  <--- Suspicious input!!!
xc=34.92, yc=18.18, r=0.37    <--- Problem!!!
```

It happens when user drags and move an existing shape, not trying to resize it. So snapping to original size is the desired behavior.

If you know why the input is weird in the first place and able to fix that, even better. But I could not figure it out.

Reply:
> It happens when user drags and move an existing shape, not trying to resize it. So snapping to original size is the desired behavior.

Ah, I missed that part. But if the selector is in move mode, it should never change the `dx`/`r` at all – this sounds rather like a bug in `bqplot`.
I could not reproduce this with either rectangle or circle, however fast I tried to grad them around – do you have any more details, or a specific dataset with which to reproduce? Otherwise, how can we be sure that a given size indicates this bug showed up – as @astrofrog noted, even very small ROI sizes can be legit?

Reply:
I cannot consistently reproduce this bug. It showed up in Imviz like this (after many tries):

1. Use https://github.com/spacetelescope/jdaviz/blob/main/notebooks/ImvizExample.ipynb
2. Load the data.
3. Draw a circle subset on image viewer. I didn't find it matter to finalize it or not; you can try it both ways.
4. Grab the drawn subset and keeps moving it around.

If you print the `selected_x` and `selected_y` out from glue-jupyter code here, you will see it eventually when the circle suddenly "disappears" but is actually super tiny.